sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/export/trace/_exporter.py

```diff
@@ -162,76 +176,142 @@ def _convert_span_to_envelope(span: Span) -> TelemetryItem:
         envelope.data = MonitorBase(
             base_data=data, base_type="RemoteDependencyData"
         )
-        if span.kind in (SpanKind.CLIENT, SpanKind.PRODUCER):
+        target = None
+        if "peer.service" in span.attributes:
+            target = span.attributes["peer.service"]
+        else:
+            if "net.peer.name" in span.attributes:
+                target = span.attributes["net.peer.name"]
+            elif "net.peer.ip" in span.attributes:
+                target = span.attributes["net.peer.ip"]
+            if "net.peer.port" in span.attributes:
+                port = span.attributes["net.peer.port"]
+                # TODO: check default port for rpc
+                # This logic assumes default ports never conflict across dependency types
+                if port != _get_default_port_http(span.attributes["http.scheme"]) and \
+                    port != _get_default_port_db(span.attributes["db.system"]):
+                    target = "{}:{}".format(target, port)
+        if span.kind is SpanKind.CLIENT:
             if "http.method" in span.attributes:  # HTTP
                 data.type = "HTTP"
-                if "net.peer.port" in span.attributes:
-                    name = ""
-                    if "net.peer.name" in span.attributes:
-                        name = span.attributes["net.peer.name"]
-                    elif "net.peer.ip" in span.attributes:
-                        name = str(span.attributes["net.peer.ip"])
-                    data.target = "{}:{}".format(
-                        name,
-                        str(span.attributes["net.peer.port"]),
-                    )
-                elif "http.url" in span.attributes:
+                scheme = span.attributes.get("http.scheme")
+                if "http.url" in span.attributes:
                     url = span.attributes["http.url"]
                     # data is the url
                     data.data = url
```

Comment:
There is some fallback logic when "http.url" is not available for data.data(httpUrl) https://github.com/microsoft/common-schema/blob/main/Mappings/OTelSemanticConvention.md#mapping-for-httpurl

Reply:
Added.